### ISSUE TITLE
A better way to save and restore position

### ISSUE THREAD

**Issue by gngshn** [tommcdo/vim-kangaroo#7]
```
diff --git a/plugin/kangaroo.vim b/plugin/kangaroo.vim
index 7ad4fda..6433735 100644
--- a/plugin/kangaroo.vim
+++ b/plugin/kangaroo.vim
@@ -2,7 +2,7 @@ function! s:push()
 	if !exists('w:positions')
 		let w:positions = []
 	endif
-	let pos = [bufnr("%"), getpos(".")]
+	let pos = [bufnr("%"), winsaveview()]
 	if len(w:positions) == 0 || w:positions[-1] != pos
 		call add(w:positions, pos)
 	endif
@@ -17,7 +17,7 @@ function! s:pop()
 	if bufnr != bufnr("%")
 		execute "edit #" . bufnr
 	endif
-	call setpos(".", pos)
+	call winrestview(pos)
 endfunction
 
 command! KangarooPush call s:push()
```

when use cscope/ctags to jump to function's definition, then I use kangaroo jump back. after that use 'j' to go to next line. the cursor will go to column 1.
If use this patch, the problem can be fix.

**Comment by tommcdo** [comment]
Looks good! Can you submit this as a pull request?

**Comment by gngshn** [comment]
ok， after I go home. I will try it.
2016年7月5日 20:33，"Tom McDonald" <notifications@github.com>写道：

**Comment by tommcdo** [comment]
Fixed by #9. Thanks again!
